### PR TITLE
Port make-temp-name

### PR DESCRIPTION
**Pull request by yshalenyk**
Implements #1223

### CODE REVIEW

**Comment by yshalenyk** [comment]
Seems like I still have some formatting issues, going to fix them also...

File: rust_src/src/fileio.rs [hunk 1]
```diff
@@ -167,4 +168,25 @@ pub fn find_file_name_handler(filename: LispStringRef, operation: LispObject) ->
...
+/// Generate temporary file name (string) starting with PREFIX (a string).
+///
+/// This function tries to choose a name that has no existing file.
+/// For this to work, PREFIX should be an absolute file name, and PREFIX
+/// and the returned string should both be non-magic.
+///
+/// There is a race condition between calling `make-temp-name' and
+/// later creating the file, which opens all kinds of security holes.
+/// For that reason, you should normally use `make-temp-file' instead.
+#[lisp_fn(min = "1")]
```
Comment (brotzeit) [inline rust_src/src/fileio.rs]: No need for `min` here.

**Comment by yshalenyk** [comment]
Should I squash commits into one?

**Comment by brotzeit** [comment]
Not necessary. That's pretty convenient with github.

**Pull request merged**